### PR TITLE
#102: added hash-history-router

### LINKED ISSUE #102: Support for createHashHistory
I'm interested in wanting to use this library for some projects I'm involved in moving forward. It looks like it currently only uses history's `createBrowserHistory` with HTML5 history. There are some situations beyond my control where I need to use hash routes, which is holding me back from jumping head first into this library. Also I'm not sure about the implementation details but I'm assuming with `createMemoryHistory` this could also be used with React Native?

### PR DESCRIPTION
**Pull request by adamshone**
Fixes #102

### CODE REVIEW

**Comment by michaelhogg** [comment]
Many thanks for this @adamshone :thumbsup:

Doing a diff between these pairs of files:

* `/src/browser-router.js` :arrow_right: `/src/hash-history-router.js`
* `/test/browser-router.spec.js` :arrow_right: `/test/hash-history-router.spec.js`

I see your proposed new HashHistoryRouter is basically a copy & paste of the existing BrowserRouter, with support for the [`hashType`](https://github.com/mjackson/history/blob/v4.4.0/README.md#modifying-the-hash-type-in-createhashhistory) arg for `createHashHistory`.

Hope to see this merged and released soon!

File: src/hash-history-router.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  passRouterStateToReducer = false
+}: HashHistoryRouterArgs) => {
+  const history = useBasename(useQueries(createHashHistory))({
+    basename,
+    hashType
+  });
+
+  const { pathname, search } = getLocation();
+  const location = history
+    .createLocation({ pathname, search });
```
Comment (michaelhogg) [inline src/hash-history-router.js]: :warning: These parameters for `createLocation()` only work correctly for the BrowserRouter.

For this HashHistoryRouter, the `hash` property from `getLocation()` needs to be used.

I've made a fix here which seems to work correctly for me:

[**Fix bug: incorrect initial location when installing router**](https://github.com/michaelhogg/redux-little-router/commit/977964a)
```js
const { hash } = getLocation();

// The hash will begin with one of the following prefixes:
//     #     If hashType == 'noslash'
//     #/    If hashType == 'slash'
//     #!/   If hashType == 'hashbang'

const hashWithoutPrefix = hash.replace(/^((#\/)|(#!\/)|(#))/, '');

const pathname = `/${ hashWithoutPrefix }`;

const location = history.createLocation({ pathname });
```

**Comment by adamshone** [comment]
Thanks @michaelhogg, looks good to me

**Comment by xMartin** [comment]
How can I help to get this merged?

**Comment by belohlavek** [comment]
Same,.whats missing?

**Comment by tptee** [comment]
Released hash history support in `v14.0.0-3`!

**Pull request closed without merging**